launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -82,8 +83,12 @@
   /** Used internally to create unique logger names. */
   private static final AtomicInteger COUNTER = new AtomicInteger();
 
+  public static final ThreadFactory REDIRECTOR_FACTORY = new NamedThreadFactory("launcher-proc-%d");
```

Comment:
How should this be shared do you think without making it public static in either SparkLauncher or ChildProcApphandle?